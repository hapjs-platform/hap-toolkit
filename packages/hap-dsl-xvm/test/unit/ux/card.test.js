/*
 * Copyright (c) 2021-present, the hapjs-platform Project Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

const path = require('@jayfate/path')
const { wipeDynamic } = require('hap-dev-utils')
const { resolveEntries, compileFiles } = require('../../utils')

/**
 * Component
 */
describe('Card 编译测试', () => {
  it('compile-card', async () => {
    // const basedir = path.resolve(__dirname, '../../case/ux/')
    // const entries = resolveEntries(basedir, 'TestCard')

    // const stats = await compileFiles(entries)
    // const json = stats.toJson({source: true})

    // json.modules.forEach(module => {
    //   expect(wipeDynamic(module.source)).toMatchSnapshot(module.id)
    // })
  })
})
