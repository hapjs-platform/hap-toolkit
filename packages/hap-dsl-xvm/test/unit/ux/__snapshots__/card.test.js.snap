// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card 编译测试 compile-card: ../../../lib/loaders/template-loader.js!../../../lib/loaders/fragment-loader.js?index=0&type=template!./TestCard/Card1/testcard1.ux?uxType=card& 1`] = `
"module.exports = {
  \\"type\\": \\"div\\",
  \\"attr\\": {},
  \\"children\\": [
    {
      \\"type\\": \\"list\\",
      \\"attr\\": {},
      \\"children\\": [
        {
          \\"type\\": \\"list-item\\",
          \\"attr\\": {
            \\"type\\": \\"item\\"
          }
        }
      ]
    },
    {
      \\"type\\": \\"refresh\\",
      \\"attr\\": {}
    },
    {
      \\"type\\": \\"richtext\\",
      \\"attr\\": {
        \\"type\\": \\"ux\\"
      }
    },
    {
      \\"type\\": \\"swiper\\",
      \\"attr\\": {},
      \\"children\\": [
        {
          \\"type\\": \\"text\\",
          \\"attr\\": {}
        }
      ]
    },
    {
      \\"type\\": \\"tabs\\",
      \\"attr\\": {},
      \\"children\\": [
        {
          \\"type\\": \\"tab-bar\\",
          \\"attr\\": {}
        },
        {
          \\"type\\": \\"tab-content\\",
          \\"attr\\": {}
        }
      ]
    },
    {
      \\"type\\": \\"slider\\",
      \\"attr\\": {}
    },
    {
      \\"type\\": \\"video\\",
      \\"attr\\": {
        \\"src\\": \\"\\"
      }
    },
    {
      \\"type\\": \\"web\\",
      \\"attr\\": {}
    }
  ]
}"
`;

exports[`Card 编译测试 compile-card: ../../../lib/loaders/template-loader.js!../../../lib/loaders/fragment-loader.js?index=0&type=template!./TestCard/Card2/testcard2.ux?uxType=card& 1`] = `
"module.exports = {
  \\"type\\": \\"div\\",
  \\"attr\\": {},
  \\"children\\": [
    {
      \\"type\\": \\"div\\",
      \\"attr\\": {}
    },
    {
      \\"type\\": \\"popup\\",
      \\"attr\\": {
        \\"target\\": \\"test_popup\\",
        \\"placement\\": \\"top\\"
      },
      \\"children\\": [
        {
          \\"type\\": \\"text\\",
          \\"attr\\": {
            \\"value\\": \\"popup\\"
          }
        }
      ]
    },
    {
      \\"type\\": \\"stack\\",
      \\"attr\\": {},
      \\"children\\": [
        {
          \\"type\\": \\"text\\",
          \\"attr\\": {
            \\"value\\": \\"1\\"
          }
        },
        {
          \\"type\\": \\"text\\",
          \\"attr\\": {
            \\"value\\": \\"2\\"
          }
        }
      ]
    },
    {
      \\"type\\": \\"list\\",
      \\"attr\\": {},
      \\"children\\": [
        {
          \\"type\\": \\"block\\",
          \\"attr\\": {},
          \\"children\\": [
            {
              \\"type\\": \\"list-item\\",
              \\"attr\\": {
                \\"type\\": \\"type1\\"
              }
            },
            {
              \\"type\\": \\"list-item\\",
              \\"attr\\": {
                \\"type\\": \\"type2\\"
              }
            }
          ]
        }
      ]
    },
    {
      \\"type\\": \\"image\\",
      \\"attr\\": {
        \\"src\\": \\"\\"
      }
    },
    {
      \\"type\\": \\"progress\\",
      \\"attr\\": {}
    },
    {
      \\"type\\": \\"rating\\",
      \\"attr\\": {}
    },
    {
      \\"type\\": \\"text\\",
      \\"attr\\": {},
      \\"children\\": [
        {
          \\"type\\": \\"span\\",
          \\"attr\\": {}
        }
      ]
    },
    {
      \\"type\\": \\"a\\",
      \\"attr\\": {
        \\"href\\": \\"\\"
      }
    },
    {
      \\"type\\": \\"input\\",
      \\"attr\\": {
        \\"type\\": \\"text\\"
      }
    },
    {
      \\"type\\": \\"label\\",
      \\"attr\\": {}
    },
    {
      \\"type\\": \\"picker\\",
      \\"attr\\": {
        \\"type\\": \\"text\\"
      }
    },
    {
      \\"type\\": \\"select\\",
      \\"attr\\": {},
      \\"children\\": [
        {
          \\"type\\": \\"option\\",
          \\"attr\\": {
            \\"value\\": \\"\\"
          }
        }
      ]
    },
    {
      \\"type\\": \\"switch\\",
      \\"attr\\": {}
    },
    {
      \\"type\\": \\"textarea\\",
      \\"attr\\": {}
    }
  ]
}"
`;

exports[`Card 编译测试 compile-card: ../../../lib/loaders/template-loader.js!../../../lib/loaders/fragment-loader.js?index=0&type=template!./TestCard/Events1/testevents1.ux?uxType=card& 1`] = `
"module.exports = {
  \\"type\\": \\"div\\",
  \\"attr\\": {},
  \\"children\\": [
    {
      \\"type\\": \\"div\\",
      \\"attr\\": {},
      \\"events\\": {
        \\"click\\": \\"click\\"
      }
    },
    {
      \\"type\\": \\"div\\",
      \\"attr\\": {},
      \\"events\\": {
        \\"longpress\\": \\"longpress\\"
      }
    },
    {
      \\"type\\": \\"input\\",
      \\"attr\\": {},
      \\"events\\": {
        \\"focus\\": \\"focus\\"
      }
    },
    {
      \\"type\\": \\"input\\",
      \\"attr\\": {},
      \\"events\\": {
        \\"blur\\": \\"blur\\"
      }
    }
  ]
}"
`;

exports[`Card 编译测试 compile-card: ../../../lib/loaders/template-loader.js!../../../lib/loaders/fragment-loader.js?index=0&type=template!./TestCard/Events2/testevents2.ux?uxType=card& 1`] = `
"module.exports = {
  \\"type\\": \\"div\\",
  \\"attr\\": {},
  \\"children\\": [
    {
      \\"type\\": \\"div\\",
      \\"attr\\": {},
      \\"events\\": {
        \\"apper\\": \\"appear\\"
      }
    },
    {
      \\"type\\": \\"div\\",
      \\"attr\\": {},
      \\"events\\": {
        \\"disapper\\": \\"disappear\\"
      }
    },
    {
      \\"type\\": \\"input\\",
      \\"attr\\": {},
      \\"events\\": {
        \\"swipe\\": \\"onswipe\\"
      }
    }
  ]
}"
`;

exports[`Card 编译测试 compile-card: ./TestCard/Card1/testcard1.ux?uxType=card 1`] = `
"
var $app_style$ = {}
var $app_script$ = null
$app_define$('@app-component/testcard1', [], function($app_require$, $app_exports$, $app_module$) {
    $app_module$.exports.template = require(\\"!../../../../../lib/loaders/template-loader.js!../../../../../lib/loaders/fragment-loader.js?index=0&type=template!./testcard1.ux?uxType=card&\\")
});
$app_bootstrap$('@app-component/testcard1',{ packagerVersion: QUICKAPP_TOOLKIT_VERSION });"
`;

exports[`Card 编译测试 compile-card: ./TestCard/Card2/testcard2.ux?uxType=card 1`] = `
"
var $app_style$ = {}
var $app_script$ = null
$app_define$('@app-component/testcard2', [], function($app_require$, $app_exports$, $app_module$) {
    $app_module$.exports.template = require(\\"!../../../../../lib/loaders/template-loader.js!../../../../../lib/loaders/fragment-loader.js?index=0&type=template!./testcard2.ux?uxType=card&\\")
});
$app_bootstrap$('@app-component/testcard2',{ packagerVersion: QUICKAPP_TOOLKIT_VERSION });"
`;

exports[`Card 编译测试 compile-card: ./TestCard/Events1/testevents1.ux?uxType=card 1`] = `
"
var $app_style$ = {}
var $app_script$ = null
$app_define$('@app-component/testevents1', [], function($app_require$, $app_exports$, $app_module$) {
    $app_module$.exports.template = require(\\"!../../../../../lib/loaders/template-loader.js!../../../../../lib/loaders/fragment-loader.js?index=0&type=template!./testevents1.ux?uxType=card&\\")
});
$app_bootstrap$('@app-component/testevents1',{ packagerVersion: QUICKAPP_TOOLKIT_VERSION });"
`;

exports[`Card 编译测试 compile-card: ./TestCard/Events2/testevents2.ux?uxType=card 1`] = `
"
var $app_style$ = {}
var $app_script$ = null
$app_define$('@app-component/testevents2', [], function($app_require$, $app_exports$, $app_module$) {
    $app_module$.exports.template = require(\\"!../../../../../lib/loaders/template-loader.js!../../../../../lib/loaders/fragment-loader.js?index=0&type=template!./testevents2.ux?uxType=card&\\")
});
$app_bootstrap$('@app-component/testevents2',{ packagerVersion: QUICKAPP_TOOLKIT_VERSION });"
`;
